crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -437,25 +449,37 @@ async fn parse_general_onchain_order_placement_data(
             let (order_data, owner, signing_scheme, order_uid) =
                 extract_order_data_from_onchain_order_placement_event(&event, domain_separator)?;
 
-            let quote = get_quote(quoter, order_data, signing_scheme, &event, &quote_id).await?;
+            let quote_result =
+                get_quote(quoter, order_data, signing_scheme, &event, &quote_id).await;
+            let mut placement_error = None;
+            let mut quote = None;
+            match quote_result.clone() {
+                Ok(found_quote) => quote = Some(found_quote),
+                Err(err) => placement_error = Some(err),
+            }
```

Comment:
uber-nano-nit: A slightly more idiomatic way of writing this would be:
```suggestion
            let (quote, placement_error) = match quote_result.clone() {
                Ok(found_quote) => (Some(found_quote), None),
                Err(err) => (None, Some(err)),
            };
```